examples/08_interpolation_join.py

```diff
@@ -0,0 +1,179 @@
+    layout="compressed",
+)
+for ax, col in zip(axes.ravel(), ["TMAX", "PRCP", "SNOW"]):
+    ax.scatter(
+        join[col].values,
+        join[f"{col}_predicted"].values,
+        alpha=0.1,
+    )
+    ax.set_aspect(1)
+    ax.set_xlabel(f"true {col}")
+    ax.set_ylabel(f"predicted {col}")
+
+######################################################################
+# We see that in this case the interpolation join works well for the temperature, but not precipitation nor snow.
+# So we will only add the temperature to our flights table.
+
+aux_table = aux_table.drop(["PRCP", "SNOW"], axis=1)
+
+######################################################################
+# Loading the flights table
+# -------------------------
+# We load the flights table and join it to the airports table using the flights’ "Origin" which refers to the departure airport’s IATA code.
+# We use only a subset to speed up the example.
+
+flights = fetch_figshare("41771418").X[["Year_Month_DayofMonth", "Origin", "ArrDelay"]]
+flights = flights.sample(20_000, random_state=0, ignore_index=True)
+airports = fetch_figshare("41710257").X.loc[
+    :, ["iata", "airport", "state", "lat", "long"]
+]
+flights = flights.merge(airports, left_on="Origin", right_on="iata")
+# printing the first row is more readable than the head() when we have many columns
+flights.iloc[0]
+
+######################################################################
+# Joining the flights and weather data
+# ------------------------------------
+# As before, we initialize our join transformer with the weather table.
+# Then, we use it to transform the flights table -- it adds a "TMAX" column containing the predicted maximum daily temperature.
+#
+
+joiner = InterpolationJoin(
+    aux_table,
+    main_key=["lat", "long", "Year_Month_DayofMonth"],
+    aux_key=["LATITUDE", "LONGITUDE", "YEAR/MONTH/DAY"],
+)
+join = joiner.fit_transform(flights)
+join.head()
+
+######################################################################
+# Sanity checks
+# -------------
+# This time we do not have a ground truth for the temperatures.
+# We can perform a few basic sanity checks.
+
+state_temperatures = join.groupby("state")["TMAX"].mean().sort_values()
+
+######################################################################
+# states with the lowest average predicted temperatures: Alaska, Montana, North Dakota, Washington, Minnesota
+state_temperatures.head()
+
+######################################################################
+# states with the highest predicted temperatures: Puerto Rico, Virgin Islands, Hawaii, Florida, Louisiana
+state_temperatures.tail()
+
+######################################################################
+# Higher latitudes (farther up north) are colder -- the airports in this dataset are in the United States.
+fig, ax = plt.subplots()
+ax.scatter(join["lat"], join["TMAX"])
+ax.set_xlabel("Latitude (higher is farther north)")
+ax.set_ylabel("TMAX")
```

Comment:
To remove the useless matplotlib print
```suggestion
ax.set_ylabel("TMAX");
```

Reply:
I like it; the pre-commit formatting style check doesn't

Reply:
Ah, annoying. Should we remove this check from flake8 (or black) for the example section?

Reply:
+1 for removing the check -- I think your suggestion makes the rendered example nicer so it would be disappointing to forgo it to make a linter happy. In general I wonder if the style checks aren't a touch too aggressive; for example your fix #754 had to work around a limitation of the linter, not fix a problem in skrub

Reply:
Yes, there is a discussion we need to have around linting. https://github.com/skrub-data/skrub/issues/755 is a first step in this direction.

Reply:
I saw other examples call plt.show after making figures; I did this too and therefore the cell's output is now None